File: bench/template/math/fib_rec/fib_rec.py
import json
import time

def fib(n):
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)

n = {{ .N }}

start = time.perf_counter()
result = fib(n)
duration = (time.perf_counter() - start) * 1000

print(json.dumps({
    "duration_ms": duration,
    "output": result,
}))

File: bench/template/math/fib_rec/fib_rec.ts
function fib(n: number): number {
    if (n <= 1) return n;
    return fib(n - 1) + fib(n - 2);
}

const n = {{ .N }};
const start = performance.now();
const result = fib(n);
const duration = performance.now() - start;

const output = {
    duration_ms: duration,
    output: result,
};

console.log(JSON.stringify(output));
